tests/debug_api.rs

```diff
@@ -0,0 +1,120 @@
+#![cfg(feature = "testing")]
```

Comment:
I think something that is missing from this test is also the RLP decoding. We should RLP decode the raw transaction and check that it matches the transaction we query using `eth_getTransactionByHash`.